### ISSUE TITLE
Indent rule is based on the indentation of the first line of code

### ISSUE THREAD

**Issue by darkred** [eslint/eslint#6993]
**What version of ESLint are you using?**
ESlint 3.4.0

**What parser (default, Babel-ESLint, etc.) are you using?**
**Please show your full configuration:**
`/*eslint indent: ["warn", "tab"]*/`

**What did you do? Please include the actual source code causing the issue.**
Consider this simple  code:
```
/*eslint indent: ["warn", "tab"]*/
	var a = 1;
alert(a);
```
**What did you expect to happen?**
IMHO, unless there's a special option in the `indent` rule to configure otherwise,
all lines should be expected to have at least no indentation (=0 tabs),
so ESLint should show warning for line 2 (`Expected indentation of 0 tab character but found 1 `) 
not 3.

**What actually happened? Please include the actual, raw output from ESLint.**
Because the 1st code line has 1 tab indentation, 
all code lines are expected to have at least 1 tab indentation,
so ESLint shows:
```
  3:1  warning  Expected indentation of 1 tab character but found 0       indent

✖ 1 problem (0 errors, 1 warning)
```

**Comment by gyandeeps** [comment]
related/duplicate https://github.com/eslint/eslint/issues/3737

**Comment by platinumazure** [comment]
Closing as duplicate, please follow the linked issue. Thanks!
